tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/date_and_time.py

```diff
@@ -21,3 +29,59 @@ def __init__(self, marionette):
     def select_time_format(self, time_format):
         self.marionette.find_element(*self._24h_selector_locator).tap()
         self.select(time_format)
+
+    def set_automatic_time_update(self, state):
+        """
+        toggles 'Set Time Automatically' flag
+        """
+
+        Wait(self.marionette).until(expected.element_displayed(
+            Wait(self.marionette).until(
+                expected.element_present(*self._autotime_enabled_locator))))
+
+        if not state:
+            if self.is_autotime_enabled is True:
```

Comment:
Having this `if` on `state` looks fine to me. That's ` if self.is_autotime_enabled is True ` which is the problem to me.

That is to say, it can lead to a false positive. For instance: The tests deactivates autotime, but due to a kernel bug, autotime was already deactivated, so the test won't report an error until you try to activate autotime 10 lines later in the test file. This will make the test hard to debug or worse, if a test doesn't try to activate autotime, lead to an actual false positive.